proposal/package-importer.d.ts.md

```diff
@@ -0,0 +1,386 @@
+  - If `resolved` has the scheme `file:` and an extension of `sass`, `scss` or
+    `css`, return it.
+  - Otherwise, if `resolved` is not null, throw an error.
+- Let `resolved` be the result of [Resolving package exports] to resolve `fullPath`
+  with the `style` condition set, using `packageManifest`.
+  - If `resolved` has the scheme `file:` and an extension of `sass`, `scss` or
+    `css`, return it.
+  - Otherwise, if `resolved` is not null, throw an error.
+- If `subPath` is empty, return result of [resolving package root values].
+- Let `resolved` be `subPath` resolved relative to `packageRoot`.
+- Return the result of [resolving a file url] with `resolved`.
+
+
+[previous url]: ../accepted/prev-url.d.ts.md
+[Resolving package exports]: #resolving-package-exports
+[resolving package root values]: #resolving-package-root-values
+[resolving a package name]: #resolving-a-package-name
+[parsing a url]: https://url.spec.whatwg.org/#concept-url-parser
+[resolving the root directory for a package]:
+    #resolving-the-root-directory-for-a-package
+[resolving a file url]: ../spec/modules.md#resolving-a-file-url
+
+### Resolving package root values
+
+This algorithm takes a string `packagePath` which is the root directory for a
+package and `packageManifest`, which is the contents of the `package.json` file,
+and returns a file URL.
+
+- Let `sassValue` be the value of `sass` in `packageManifest`.
+- If `sassValue` is a relative path with an extension of `sass`, `scss` or
+  `css`, return the `packagePath` appended with `sassValue`.
+- Let `styleValue` be the value of `style` in `packageManifest`.
+- If `styleValue` is a relative path with an extension of `css`, return the
+  `packagePath` appended with `styleValue`.
+- Otherwise return the result of [resolving a file url] with `packagePath`.
+
+[resolving the root directory for a package]:
+    #resolving-the-root-directory-for-a-package
+[resolving a file url]: ../spec/modules.md#resolving-a-file-url
+
+### Resolving a package name
+
+This algorithm takes a string, `url`, and returns the portion that identifies
+the Node package.
+
+- If `url` starts with `@`, it is a scoped package. Return the first 2 [url path
+  segments], including the separating `/`
+- Otherwise, return the first url path segment.
+
+[url path segments]: https://url.spec.whatwg.org/#url-path-segment
+
+### Resolving the root directory for a package
+
+This algorithm takes a string, `packageName`, an absolute URL
+`currentDirectory`, and an optional absolute URL `previousUrl`, and returns an
+absolute URL to the root directory for the most proximate installed
+`packageName`.
+
+> We need to replicate Node's behavior, as defined in [Loading from node_modules
+> folders], of walking up the directory chain to find packages from
+> `previousUrl`, so that packages can use the correct version of the packages
+> that they depend on.
+
+- If `previousUrl` is null, return the result of `require.resolve` with the
```

Comment:
If `require.resolve` is a specified procedure, link to it here. If it's not, I'm afraid we'll need to write out the behavior ourselves, since this will need to be implemented in non-Node.js environments.